data_dictionary/data_dictionary.hh

```diff
@@ -125,6 +125,7 @@ public:
     const db::extensions& extensions() const;
     const gms::feature_service& features() const;
     replica::database& real_database() const; // For transition; remove
+    replica::database* real_database_ptr() const;
```

Comment:
Why do we need the real_database_ptr thing?  The comment says real_database() is fo transition. It will be removed.

Reply:
It's tempting to use `needs_repair_before_gc` for rf/locality check and this function gets `replica::database`. Also, I didn't know how to retrieve replication factor (or `token_metadata` to get rf) out of `data_dictionary::database`.

`real_database()` wasn't enough as it throws `std::bad_function_call` for the classes deriving from `data_dictionary_impl` which are used in boost tests

Reply:
hmm, ok.